project/BRANCHES-AND-TAGS.md

```diff
@@ -0,0 +1,35 @@
+Branches and tags
+=================
+
+Note: details of the release process for the Engine are documented in the
+[RELEASE-CHECKLIST](https://github.com/docker/docker/blob/master/project/RELEASE-CHECKLIST.md).
+
+# Branches
+
+Docker projects should normally have only three living branches at all time, including the regular
+`master` branch:
+
+## `docs` branch
```

Comment:
I would also rather the docs be published from the already existing release branch - having them separate generally means the code in that branch does not get updated when point releases happen (and we don't do docs point release branches) and by having them separate, both the docs and release branches are not fully up todate.

However - this PR is about documenting what we do today :) so this change would go elsewhere.